Remove HTMLMediaElement.webkitHasClosedCaptions/webkitClosedCaptionsVisible

Ahmad-S792 opened the pull request:
#### 0a73c4dbb7a070ffeac5aa344cfc728cefc86a3e
<pre>
Remove HTMLMediaElement.webkitHasClosedCaptions/webkitClosedCaptionsVisible

<a href="https://bugs.webkit.org/show_bug.cgi?id=250870">https://bugs.webkit.org/show_bug.cgi?id=250870</a>
rdar://problem/104451148

Reviewed by Eric Carlson.

Inspired by: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=159317

This patch removes &apos;prefixed&apos; closedCaption extensions and align with Web-Spec.

* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(HTMLMediaElement::HTMLMediaElement): Remove unused &apos;m_webkitLegacyClosedCaptionOverride&apos;
(HTMLMediaElement::configureTextTrackGroup): Remove &apos;condition&apos;
(HTMLMediaElement::setWebkitClosedCaptionsVisible): Removed
(HTMLMediaElement::webkitClosedCaptionsVisible): Removed
(HTMLMediaElement::webkitHasClosedCaptions): Removed
(HTMLMediaElement::captionPreferencesChanged): Removed prefixed reference
* Source/WebCore/html/HTMLMediaElement.idl: Remove &apos;FIXME&apos; and legacy stuff
* Source/WebCore/page/CaptionUserPreferences.cpp:
(CaptionUserPreferences::textTrackSelectionScore): Remove use of &apos;legacyOverride&apos; and uses
* Source/WebKitLegacy/mac/DOM/DOMHTMLMediaElement.mm: Make WebKit Legacy use &apos;un-prefixed&apos; extensions
* ManualTests/media-controls.html: Rebaselined
* LayoutTests/media/track/track-in-band-legacy-api.html: Deleted
* LayoutTests/media/track/track-in-band-legacy-api-expected.txt: Deleted
* LayoutTests/platform/mac-wk1/TestExpectations: Remove &apos;deleted&apos; Test Expectations
* LayoutTests/platform/glib/TestExpectations: Ditto
* LayoutTests/media/media-captions.html: Deleted
* LayoutTests/media/media-captions-expected.txt: Deleted
* LayoutTests/media/track/track-legacyapi-with-automatic-mode-expected.txt: Deleted
* LayoutTests/media/track/track-legacyapi-with-automatic-mode.html: Deleted

Canonical link: <a href="https://commits.webkit.org/267780@main">https://commits.webkit.org/267780@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/efe2bd5204196ed2def465dd2af032b252738340

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/17616 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/17941 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18471 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19425 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16474 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/17809 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21227 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18087 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/18572 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/17827 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18128 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15295 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20269 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15363 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16040 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/22636 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16372 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16209 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20492 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/16780 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14213 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/15897 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4211 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20263 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/16623 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 commented on Source/WebCore/html/HTMLMediaElement.cpp:
```diff
@@ -4939,11 +4938,6 @@ void HTMLMediaElement::configureTextTrackGroup(const TrackGroup& group)
     if (trackToEnable) {
         trackToEnable->setMode(TextTrack::Mode::Showing);
-
-        // If user preferences indicate we should always display captions, make sure we reflect the
```
@eric-carlson - can you share your insights for following?

eric-carlson replied:
I don't think it is an issue.

Ahmad-S792 replied:
Thanks!

Ahmad-S792 committed:
Remove HTMLMediaElement.webkitHasClosedCaptions/webkitClosedCaptionsVisible

https://bugs.webkit.org/show_bug.cgi?id=250870
rdar://problem/104451148

Reviewed by Eric Carlson.

Inspired by: https://src.chromium.org/viewvc/blink?view=revision&revision=159317

This patch removes 'prefixed' closedCaption extensions and align with Web-Spec.

* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/html/HTMLMediaElement.cpp:
(HTMLMediaElement::HTMLMediaElement): Remove unused 'm_webkitLegacyClosedCaptionOverride'
(HTMLMediaElement::configureTextTrackGroup): Remove 'condition'
(HTMLMediaElement::setWebkitClosedCaptionsVisible): Removed
(HTMLMediaElement::webkitClosedCaptionsVisible): Removed
(HTMLMediaElement::webkitHasClosedCaptions): Removed
(HTMLMediaElement::captionPreferencesChanged): Removed prefixed reference
* Source/WebCore/html/HTMLMediaElement.idl: Remove 'FIXME' and legacy stuff
* Source/WebCore/page/CaptionUserPreferences.cpp:
(CaptionUserPreferences::textTrackSelectionScore): Remove use of 'legacyOverride' and uses
* Source/WebKitLegacy/mac/DOM/DOMHTMLMediaElement.mm: Make WebKit Legacy use 'un-prefixed' extensions
* ManualTests/media-controls.html: Rebaselined
* LayoutTests/media/track/track-in-band-legacy-api.html: Deleted
* LayoutTests/media/track/track-in-band-legacy-api-expected.txt: Deleted
* LayoutTests/platform/mac-wk1/TestExpectations: Remove 'deleted' Test Expectations
* LayoutTests/platform/glib/TestExpectations: Ditto
* LayoutTests/media/media-captions.html: Deleted
* LayoutTests/media/media-captions-expected.txt: Deleted
* LayoutTests/media/track/track-legacyapi-with-automatic-mode-expected.txt: Deleted
* LayoutTests/media/track/track-legacyapi-with-automatic-mode.html: Deleted

Canonical link: https://commits.webkit.org/267780@main
